src/nxt_openssl.c

```diff
@@ -1781,7 +1785,11 @@ nxt_openssl_copy_error(u_char *p, u_char *end)
     clear = 0;
 
     for ( ;; ) {
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
```

Comment:
This can even be used with older version (tested with 1.1.1)

Reply:
Yes; please use it for all cases and remove the conditional.

That also allows removing the pragma, I guess.

Reply:
The [migration guide](https://www.openssl.org/docs/man3.0/man7/migration_guide.html) said there's few variants


> ERR_load_*(), ERR_func_error_string(), ERR_get_error_line(), ERR_get_error_line_data(), ERR_get_state()
>> OpenSSL now loads error strings automatically so these functions are not needed.

> Applications should use [ERR_get_error_all(3)](https://www.openssl.org/docs/man3.0/man3/ERR_get_error_all.html), or pick information with ERR_peek functions and finish off with getting the error code by using [ERR_get_error(3)](https://www.openssl.org/docs/man3.0/man3/ERR_get_error.html).

Reply:
Sorry, my mistake, this is really openssl 3 only, see

https://www.openssl.org/docs/man3.0/man3/ERR_get_error_line_data.html   (ERR_get_error_all is there)
https://www.openssl.org/docs/man1.1.1/man3/ERR_get_error_line_data.html (ERR_get_error_all not available)